### PR TITLE
Follow-up for comments on #15891

### PR DESCRIPTION
**Pull request by bluca**
Apply suggestions, and fix https://github.com/systemd/systemd/issues/16484

### CODE REVIEW

File: src/basic/mkdir.c [hunk 1]
```diff
@@ -136,7 +136,7 @@ int mkdir_parents_internal(const char *prefix, const char *path, mode_t mode, ui
                 if (prefix && path_startswith(prefix, t))
                         continue;
 
-                if (uid == UID_INVALID && gid == UID_INVALID && flags == 0) {
+                if (!uid_is_valid(uid) && !uid_is_valid(gid) && flags == 0) {
```
Comment (poettering) [inline src/basic/mkdir.c]: gid_is_valid for the gid check, please

File: src/basic/mkdir.c [hunk 2]
```diff
@@ -167,7 +167,7 @@ int mkdir_p_internal(const char *prefix, const char *path, mode_t mode, uid_t ui
         if (r < 0)
                 return r;
 
-        if (uid == UID_INVALID && gid == UID_INVALID && flags == 0) {
+        if (!uid_is_valid(uid) && !uid_is_valid(gid) && flags == 0) {
```
Comment (poettering) [inline src/basic/mkdir.c]: ditto

File: src/basic/mkdir.c (hunk 1 above)
Comment (bluca) [reply src/basic/mkdir.c]: thanks, fixed

**Comment by poettering** [comment]
thanks for the quick fix! lgtm!

**Pull request merged**